content/docs/05.developer-guide/03.scripts.md

```diff
@@ -0,0 +1,800 @@
+
+### Add `git.Clone` as the first task in a `WorkingDirectory`
+
+Adding the `git.Clone` task directly as the first child task of the `WorkingDirectory` task will ensure that you clone your repository into an empty directory before any other task would generate any output artifacts.
+
+### Private Git repositories
+Typically, you would want to use `git.Clone`  with a private GitHub repository. Make sure to:
+1. Add your organization/user name as `username`
+2. Generate your access token and provide it on the `password` property.
+
+Below you can find links to instructions on how to generate an access token for the relevant Git platform:
+
+- [GitHub](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)
+- [Gitlab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html)
+- [Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/create-a-repository-access-token/)
+- [AWS CodeCommit](https://docs.aws.amazon.com/codecommit/latest/userguide/auth-and-access-control.html)
+- [Azure DevOps](https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows)
+
+---
+
+## When to use the `WorkingDirectory` task
+
+By default, all Kestra tasks are **stateless**. If one task generates files, those files won’t be available in downstream tasks unless they are persisted in internal storage. Upon each task completion, the temporary directory for the task is purged. This behavior is generally useful as it keeps your environment clean and dependency free, and it avoids potential privacy or security issues when exposing some data generated by a task to other processes.
+
+Despite the benefits of the stateless execution, in certain scenarios, **statefulness** is **desirable**. Imagine that you want to execute several Python scripts, and each of them generates some output data. Another script combines that data as part of an ETL/ML process. Executing those related tasks in the same working directory and **sharing state** between them is helpful for the following reasons:
+- You can **clone** your entire **GitHub branch** with multiple modules and configuration files needed to run several scripts
+- You can **execute** multiple scripts **sequentially** on the same worker or in the same container, minimizing latency,
+- **Outputs** of each task are directly available to other tasks without having to persist them within the internal storage.
+
+The `WorkingDirectory` task addresses all of these challenges. It allows you to:
+1. Clone a Git repository (or retrieve needed input files in other ways), 
+2. Run multiple tasks sequentially in the same working directory 
+3. Reuse the same file system's working directory across multiple tasks. 
+
+All tasks within the `WorkingDirectory` can use output files from previous tasks without having to use the `{{outputs.taskId.outputAttribute}}` syntax.
```

Comment:
I'm not very compfortable with __output files__ maybe use __generated fils__ as outputs has a different meaning in Kestra.

Reply:
Ill change to output artifacts -- especially given that Yann is working on displaying those in the UI